install

```diff
@@ -364,45 +379,42 @@ EOABORT
 
 ohai "Downloading and installing Homebrew..."
 Dir.chdir HOMEBREW_REPOSITORY do
-  if git
-    # we do it in four steps to avoid merge errors when reinstalling
-    system git, "init", "-q"
-
-    # "git remote add" will fail if the remote is defined in the global config
-    system git, "config", "remote.origin.url", BREW_REPO
-    system git, "config", "remote.origin.fetch", "+refs/heads/*:refs/remotes/origin/*"
+  # we do it in four steps to avoid merge errors when reinstalling
+  system git, "init", "-q"
 
-    # ensure we don't munge line endings on checkout
-    system git, "config", "core.autocrlf", "false"
+  # "git remote add" will fail if the remote is defined in the global config
+  system git, "config", "remote.origin.url", can_use_https? ? BREW_REPO : BREW_REPO_INSECURE
+  system git, "config", "remote.origin.fetch", "+refs/heads/*:refs/remotes/origin/*"
 
-    args = git, "fetch", "origin", "master:refs/remotes/origin/master",
-           "--tags", "--force"
-    system(*args)
+  # ensure we don't munge line endings on checkout
+  system git, "config", "core.autocrlf", "false"
 
-    system git, "reset", "--hard", "origin/master"
+  args = git, "fetch", "origin", "master:refs/remotes/origin/master",
+         "--tags", "--force"
+  system(*args)
 
-    system "ln", "-sf", "#{HOMEBREW_REPOSITORY}/bin/brew", "#{HOMEBREW_PREFIX}/bin/brew"
+  system git, "reset", "--hard", "origin/master"
 
-    system "#{HOMEBREW_PREFIX}/bin/brew", "update", "--force"
-  else
-    # -m to stop tar erroring out if it can't modify the mtime for root owned directories
-    # pipefail to cause the exit status from curl to propagate if it fails
-    curl_flags = "fsSL"
-    curl_flags += "k" if macos_version < "10.6"
-    core_tap = "#{HOMEBREW_PREFIX}/Homebrew/Library/Taps/homebrew/homebrew-core"
+  unless can_use_https?
+    system git, "config", "--file", "#{HOMEBREW_REPOSITORY}/.git/config", "remote.origin.url", BREW_REPO
```

Comment:
Does `brew doctor` complain if this isn't changed back?

Reply:
Moved to `brew update` in Homebrew/brew#5238.